core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -130,6 +130,16 @@ sealed trait Cause[+E] extends Product with Serializable { self =>
       case (_, Interrupt(_))    => false
     }
 
+  /**
+   * Determines if the `Cause` is a pure interruption, a `Cause` that does not
+   * contain any `Die` or `Fail` causes.
+   */
+  final def isPureInterruption: Boolean =
```

Comment:
I might call this one, `justInterrupted` or even `interruptedOnly` for greater symmetry with `interrupted: Boolean`.

Otherwise, looks fantastic!

Reply:
Good idea. Updated now.